configure.ac

```diff
@@ -1799,37 +1811,34 @@ AS_CASE([$arch,$system],
     AC_MSG_RESULT([yes])],
   [AC_MSG_RESULT([no])])
 
-## Determine if the POSIX threads library is supported
+## Determine how to link with the POSIX threads library
+
+AS_CASE([$host],
+  [*-*-mingw32|*-pc-windows],
+    [PTHREAD_LIBS="-lpthread"],
+  [AX_PTHREAD(
+    [common_cflags="$common_cflags $PTHREAD_CFLAGS"
+    saved_CFLAGS="$CFLAGS"
+    saved_LIBS="$LIBS"
+    CFLAGS="$CFLAGS $PTHREAD_CFLAGS"
+    LIBS="$LIBS $PTHREAD_LIBS"
+    AC_CHECK_FUNC([sigwait], [AC_DEFINE([HAS_SIGWAIT])])
+    LIBS="$saved_LIBS"
+    CFLAGS="$saved_CFLAGS"],
+    [AC_MSG_ERROR([POSIX threads are required but not supported on this platform])])])
+
+## Activate the systhread library
 
 AS_CASE([$enable_systhreads,$enable_unix_lib],
   [yes,no],
     [systhread_support=false
-    AC_MSG_ERROR([the Win32/POSIX threads library requires the unix library])],
+    AC_MSG_ERROR([the threads library requires the unix library])],
   [no,*|*,no],
     [systhread_support=false
-    AC_MSG_NOTICE([the Win32/POSIX threads library is disabled])],
-  [AS_CASE([$host],
-    [*-*-mingw32|*-pc-windows],
-      [systhread_support=true
-      otherlibraries="$otherlibraries systhreads"
-      PTHREAD_LIBS="-lpthread"
-      AC_MSG_NOTICE([the Win32 threads library is supported])],
-    [AX_PTHREAD(
-      [systhread_support=true
-      otherlibraries="$otherlibraries systhreads"
-      common_cflags="$common_cflags $PTHREAD_CFLAGS"
-      AC_MSG_NOTICE([the POSIX threads library is supported])
-      saved_CFLAGS="$CFLAGS"
-      saved_LIBS="$LIBS"
-      CFLAGS="$CFLAGS $PTHREAD_CFLAGS"
-      LIBS="$LIBS $PTHREAD_LIBS"
-      AC_CHECK_FUNC([sigwait], [AC_DEFINE([HAS_SIGWAIT])])
-      LIBS="$saved_LIBS"
-      CFLAGS="$saved_CFLAGS"],
-      [AS_IF([test x"$enable_systhreads" = "xyes"],
-        [AC_MSG_ERROR([the POSIX thread library is not available])],
-        [systhread_support=false
-        AC_MSG_NOTICE([the POSIX threads library is not supported])])])])])
+    AC_MSG_NOTICE([the threads library is disabled])],
+  [systhread_support=true
+  otherlibraries="$otherlibraries systhreads"
+  AC_MSG_NOTICE([the threads library is supported])])
```

Comment:
If I understand correctly, previously we were trying to guess whether systhreads can be enabled (by checking whether (1) Windows or (2) both pthreads and `sigwait` are available), and currently you ask for an explicit setting without auto-detection logic? (Or do you know assume `yes` unless `unix` is known to not be supported?)

(It's a bit sad to see the auto-detection logic go away, but I guess the people that will work on Windows support or whatever will know to just look at the 4.14 configure script to take inspiration from.)

Reply:
POSIX threads or the Win32 equivalent are now required by the runtime system (how many times will I have to write this?), so systhreads can always be built, and it's the default.  It is not built only if the user configured with `--disable-systhreads`.